core/src/main/scala/chisel3/Module.scala

```diff
@@ -107,6 +110,32 @@ object Module extends SourceInfoDoc {
   /** Returns the implicit Reset, if it is defined */
   def resetOption: Option[Reset] = Builder.currentReset
 
+  /** Returns the current implicit [[Disable]], if one is defined
+    *
+    * Note that [[Disable]] is a function of the implicit clock and reset
+    * so having no implicit clock or reset may imply no `Disable`.
+    */
+  def disable(implicit sourceInfo: SourceInfo): Option[Disable] = {
+    Builder.currentDisable match {
+      case Disable.None            => None
+      case Disable.NotHasBeenReset => hasBeenReset.map(x => autoNameRecursively("disable")(!x))
+    }
+  }
```

Comment:
The fact that this returns an `Option[Disable]` is a bit of a non-standard API for Chisel. I'm more used to APIs which throw exceptions. My question is, in what circumstances would the user not require that `disable: Some[Disable]`?

Analogously, it's similar to having `Reg(UInt(8.W))` return `Option[UInt]`. If a user is trying to instantiate a register, they likely expect to get a register and it's faster to error if they can't instantiate a register.

WDYT?

Reply:
I added `Module.clockOption` and `Module.resetOption` in https://github.com/chipsalliance/chisel/pull/3496 because the LTL APIs accept `Option[Clock]` and `Option[Bool]` (soon to be `Option[Disable]`), so we at least need the optional version for use there. I think it's reasonable to make `Module.disable` be a version that returns `Disable` and asserts if not and we could add `Module.disableOption` for use by the LTL Property APIs.

Reply:
For consistency, we can just make this `disableOption` and either add or wait until we need a `disable` that errors. Regardless, `disableOption` sounds good to me.

Reply:
Renamed `disable` to `disableOption` and added a form `disable` that asserts out if no implicit disable is defined.